x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -415,14 +391,8 @@ async function getActionInfoInternal<Source = unknown>(
     );
   }
 
-  const actionsClient = await getActionsClientWithRequest(request, source);
```

Comment:
I verified that the actions no longer throw an error with this PR but am a little confused why. The error that was thrown was `Saved object [alert/be52ab90-b383-11ed-a09e-c1c6ca06b06b] not found` which implies we're accessing the rule for the action but this is retrieving the action. I looked in the `ActionsClient.get` function and it doesn't look like we're getting the rule there either. Where is the rule being retrieved?

Reply:
Oh I read your comment on the issue and now I understand 🙈

Reply:
Yeah, it took me a while to realize `getActionsClientWithRequest` is async and does RBAC. In the future, I think we could explore moving the `ALERT_SAVED_OBJECT_TYPE` checks into alerting before calling the actions client.